src/vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient.ts

```diff
@@ -340,7 +356,30 @@ export class DataExplorerClientInstance extends Disposable {
 		profiles: Array<ColumnProfileRequest>
 	): Promise<Array<ColumnProfileResult>> {
 		return this.runBackendTask(
-			() => this._positronDataExplorerComm.getColumnProfiles(profiles, this._profileFormatOptions),
+			async () => {
+				const callbackId = generateUuid();
+				const promise = new DeferredPromise<Array<ColumnProfileResult>>();
+				this._asyncTasks.set(callbackId, promise);
+				await this._positronDataExplorerComm.getColumnProfiles(callbackId, profiles, this._profileFormatOptions);
```

Comment:
Should we also catch errors here and reject the deferred promise immediately? It will timeout after 10 seconds in any case but could be helpful for the user to have immediate feedback.

Reply:
Tracked here: https://github.com/posit-dev/positron/issues/4559
Currently the expected behavior is to catch all error and return empty results instead. We will need additional UI to somehow display these errors.